Src/DependencyCollector/Shared/Implementation/ClientServerDependencyTracker.cs

```diff
@@ -32,6 +32,18 @@ internal static DependencyTelemetry BeginTracking(TelemetryClient telemetryClien
             // but every operation must have it's own Activity and Id must be set accordingly
             // basically we repeat TelemetryClientExtensions.StartOperation here
             var activity = new Activity(DependencyActivityName);
+
+            // This is workaround for the issue https://github.com/Microsoft/ApplicationInsights-dotnet/issues/538
+            // if there is no parent Activity, ID Activity generates is not random enough to work well with 
+            // ApplicationInsights sampling algorithm
+            // This code should go away when Activity is fixed: https://github.com/dotnet/corefx/issues/18418
+            if (Activity.Current == null)
```

Comment:
It's a temporary workaround. Right way would be to solve it in Activity.

Also, are you talking about StartOperation?

It's not always possible and convenient to  call it because:
- you  don't always want to keep OperationHolder object: like in the ClientServerDependencyTracker
- you don't always want to initialize telemetry immediately: like in incoming request handling - because of delayed operation name calculation.

Also, AI don't always start activity: HttpModule does it and AI just 'patches' that activity before it is started, but does not start it